internal/services/network/subnet_resource.go

```diff
@@ -444,7 +445,7 @@ func resourceSubnetRead(d *pluginsdk.ResourceData, meta interface{}) error {
 		d.Set("enforce_private_link_service_network_policies", flattenSubnetPrivateLinkNetworkPolicy(string(props.PrivateLinkServiceNetworkPolicies)))
 
 		serviceEndpoints := flattenSubnetServiceEndpoints(props.ServiceEndpoints)
-		if err := d.Set("service_endpoints", serviceEndpoints); err != nil {
+		if err := d.Set("service_endpoints", pluginsdk.NewSet(pluginsdk.HashString, serviceEndpoints)); err != nil {
```

Comment:
fwiw we shouldn't need to set this explicitly, since we're using the default hash code here?

```suggestion
		if err := d.Set("service_endpoints", serviceEndpoints); err != nil {
```